plugin/secrets_service_account_key.go

```diff
@@ -195,6 +201,22 @@ func (b *backend) getSecretKey(ctx context.Context, s logical.Storage, rs *RoleS
 		return logical.ErrorResponse(err.Error()), nil
 	}
 
+	if ttl == 0 {
+		ttl = int(cfg.TTL.Seconds())
+	}
+	if ttl == 0 {
+		ttl = int(b.System().DefaultLeaseTTL().Seconds())
+	}
+
+	maxTTL := int(cfg.MaxTTL.Seconds())
+	if maxTTL <= 0 {
+		maxTTL = int(b.System().MaxLeaseTTL().Seconds())
+	}
+
+	if ttl > maxTTL {
+		ttl = maxTTL
+	}
```

Comment:
I don't _think_ these parts are needed.

- Vault will fall back to system Default and Max TTLs if the return secret TTLs  are <= 0
- Vault will cap the TTL at the MaxTTL automatically
- If we let Vault do that limiting (i.e. remove 216-218), then you'll get a warning response for free:
```
WARNING! The following warnings were returned from Vault:

  * TTL of "11h0m0s" exceeded the effective max_ttl of "10h0m0s"; TTL value is
  capped accordingly
```

Reply:
Yup you're right. I just verified this. Updated.